tests/pyfunc/test_model_export.py

```diff
@@ -89,17 +98,76 @@ def test_model_log(self):
                 # Remove the log directory in order to avoid adding new tests to pytest...
                 shutil.rmtree(tracking_dir)
 
-    def test_model_serve(self):
+    def _model_serve_with_conda_env(self, extra_args):
         with TempDir() as tmp:
             model_path = tmp.path("knn.pkl")
             with open(model_path, "wb") as f:
                 pickle.dump(self._knn, f)
             path = tmp.path("knn")
+
+            # create a conda yaml that installs mlflow from source in-place mode
+            conda_env_path = tmp.path("conda.yml")
+            with open(conda_env_path, "w") as f:
+                f.write("""
+                        name: mlflow
+                        channels:
+                          - defaults
+                        dependencies:
+                          - pip:
+                            - -e {}
+                        """.format(os.path.abspath(os.path.join(mlflow.__path__[0], '..'))))
+
             pyfunc.save_model(dst_path=path,
                               data_path=model_path,
                               loader_module=os.path.basename(__file__)[:-3],
                               code_path=[__file__],
+                              conda_env=conda_env_path
                               )
+            input_csv_path = tmp.path("input.csv")
+            pandas.DataFrame(self._X).to_csv(input_csv_path, header=True, index=False)
+            runner = CliRunner(env={"LC_ALL": "en_US.UTF-8", "LANG": "en_US.UTF-8"})
+            port = 5000
+
+            def runserver():
+                result = runner.invoke(mlflow.pyfunc.cli.commands,
+                                       ['serve', '--model-path', path, '--port', port] + extra_args)
+                print(result.exc_info)
+                if result.exit_code != 0:
+                    return -1
+
+            process = Process(target=runserver)
```

Comment:
If we use Popen, would not need to use multiprocessing, might simplify this a bit.

Reply:
Are you suggesting we just do Popen('mlflow pyfunc server') instead of using click.CliRunner?